MimeKit/Header.cs

```diff
@@ -312,9 +312,14 @@ protected Header (ParserOptions options, HeaderId id, string name, byte[] field,
 		/// <param name="fieldNameLength">The length of the field name (not including trailing whitespace).</param>
 		/// <param name="value">The raw value of the header.</param>
 		/// <param name="invalid"><c>true</c> if the header field is invalid; othereise, <c>false</c>.</param>
+#if NET5_0_OR_GREATER
+		[System.Runtime.CompilerServices.SkipLocalsInit]
+#endif
 		internal protected Header (ParserOptions options, byte[] field, int fieldNameLength, byte[] value, bool invalid)
 		{
-			var chars = new char[fieldNameLength];
+			Span<char> chars = fieldNameLength <= 32
+				? stackalloc char[32]
+				: new char[fieldNameLength];
```

Comment:
Same question about Span<>, but also why 32 here but 16 elsewhere?

Reply:
I did some random checking to see the averages sizes and found them to be small or large (12 bytes or 100+ bytes). Standardizing on 32 bytes seems reasonable however.